src/frontend/error.rs

```diff
@@ -0,0 +1,67 @@
+use std::{backtrace::Backtrace, fmt::Debug};
+
+use crate::middleware::{DynError, Statement, StatementTmpl};
+
+pub type FrontendResult<T, E = FrontendError> = core::result::Result<T, E>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum InnerError {
+    #[error(transparent)]
+    Tree(#[from] crate::backends::plonky2::primitives::merkletree::error::TreeError),
```

Comment:
is this one used? I've tried removing it and seems to not be used. I was trying this bcs I was thinking if maybe we could avoid using the `TreeError` since it could be 'embedded' in the `backend::Error` already.